cargo-dist/src/lib.rs

```diff
@@ -144,6 +152,88 @@ fn run_build_step(
     }
 }
 
+fn build_fake(dist_graph: &DistGraph, target: &BuildStep, manifest: &DistManifest) -> Result<()> {
+    match target {
+        // These two are the meat: don't actually run these at all, just
+        // fake them out
+        BuildStep::Generic(target) => build_fake_generic_artifacts(dist_graph, target),
+        BuildStep::Cargo(target) => build_fake_cargo_artifacts(dist_graph, target),
+        // Never run rustup
+        BuildStep::Rustup(_) => Ok(()),
+        // Copying files is fairly safe
+        BuildStep::CopyFile(CopyStep {
+            src_path,
+            dest_path,
+        }) => copy_file(src_path, dest_path),
+        BuildStep::CopyDir(CopyStep {
+            src_path,
+            dest_path,
+        }) => copy_dir(src_path, dest_path),
+        BuildStep::CopyFileOrDir(CopyStep {
+            src_path,
+            dest_path,
+        }) => copy_file_or_dir(src_path, dest_path),
+        // The remainder of these are mostly safe to run as fake steps
+        BuildStep::Zip(ZipDirStep {
+            src_path,
+            dest_path,
+            zip_style,
+            with_root,
+        }) => zip_dir(src_path, dest_path, zip_style, with_root.as_deref()),
+        BuildStep::GenerateInstaller(installer) => {
+            generate_installer(dist_graph, installer, manifest)
```

Comment:
so this one is tricky because the msi installers *can't* be built on all platforms. probably means just expanding out and applying this pattern on the match inside this function

Reply:
Aha, I didn't consider MSI - I was only testing wish shell, powershell, Homebrew.

Reply:
OK, special-cased MSI while generating real installers for the other paths.